packages/@react-spectrum/datepicker/src/styles.css

```
@@ -182,3 +182,17 @@
   padding: 0 var(--spectrum-calendar-day-padding);
   box-sizing: border-box;
 }
+
+@media (forced-colors:active) {
+  .react-spectrum-DatePicker-cell:focus {
+    forced-color-adjust: none;
+    background-color: Highlight;
+    color: HighlightText;
+  }
+  .react-spectrum-DatePicker-cell.is-read-only {
+    color: ButtonText;
+  }
+  .react-spectrum-DatePicker-cell.is-placeholder {
+    color: ButtonText;
+  }
```

Add the following, so that the placeholder text has sufficient contrast over `background-color: Highlight;`: 
```suggestion
  .react-spectrum-DatePicker-cell.is-placeholder:focus {
    color: HighlightText;
  }
}
```